### PR TITLE
[move core modules] Authenticator module to match Libra authenticator…

### PR DESCRIPTION
**Pull request by sblackshear**
… logic

This is useful for converting ed25519 public keys to authentication keys (both single and multisig) on-chain.

## Motivation

This is needed in some upcoming modules that make it easier to manage multisig keys on-chain.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

New functional tests

### CODE REVIEW

File: language/stdlib/modules/authenticator.move [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            let public_key = *Vector::borrow(public_keys, i);
+            Vector::append(
+                &mut authentication_key_preimage,
+                public_key
+            );
+            i = i + 1;
+        };
+        // TODO: add constant MULTI_ED25519_SCHEME_ID = 1u8
+        Vector::push_back(&mut authentication_key_preimage, 1u8);
+        Hash::sha3_256(authentication_key_preimage)
```
Comment (valerini) [inline language/stdlib/modules/authenticator.move:62]: Do we need to add the threshold inside the hash too?

Comment (sblackshear) [reply language/stdlib/modules/authenticator.move:62]: Good catch! I need to add some E2E tests to ensure that the expected output matches the Rust implementation...

Comment (sblackshear) [reply language/stdlib/modules/authenticator.move:64]: Fixed + added a test. Thanks for noticing this!

**Pull request closed without merging**